### DISCUSSION TITLE
Change out_filter params of PESieve_scan_ex method throw exeption when use Pe-sieve API

### DISCUSSION THREAD

**Discussion by nhat3eo22** [hasherezade/pe-sieve#113] [Q&A]
I'm building a DLL with an import Pe-sieve API to use for malware scanning. I changed the out_filter value in PEsieve_params to OUT_NO_DIR but when I run the program, it doesn't work. While not modifying this value it works properly.

![image](https://user-images.githubusercontent.com/97865127/236165794-02ba7bc0-dfa9-4498-9e75-fca192df44ba.png)

I get the exception:

<img width="875" alt="image" src="https://user-images.githubusercontent.com/97865127/236168217-88e371a0-9018-4202-b308-f146c340f690.png">

Where did I go wrong?
Thanks for your answer!

**Comment by nhat3eo22** [comment]
My code is simple:
<img width="599" alt="image" src="https://user-images.githubusercontent.com/97865127/236169804-19b47ac4-91cf-4445-b6ee-2c5e12601205.png">

**Reply by hasherezade** [reply]
ok, I confirmed that the error you encountered was already fixed (https://github.com/hasherezade/pe-sieve/commit/a2e917b163dc4ea49a8b2e5280e80f40c890ca1a). The fix will be available in the upcoming release. For now you can use the PE-sieve built from the `master`, or a testbuild that I prepared for you:

[pe-sieve64_dll.zip](https://github.com/hasherezade/pe-sieve/files/11398461/pe-sieve64_dll.zip)

please let me know if it solves your problem!

**Reply by nhat3eo22** [reply]
Thanks for your answer, it works fine, looking forward to a new release from you. Thanks have a nice day!

**Reply by hasherezade** [reply]
@nhat3eo22 - the new release including the fix is ready: https://github.com/hasherezade/pe-sieve/releases/tag/v0.3.6 - please check it out!

**Reply by hasherezade** [reply]
@nhat3eo22 - feel free to marked this question as answered once you make sure that everything works fine :)

**Comment by hasherezade** [comment]
hi @nhat3eo22 ! Thanks for reporting, I reproduced the error, and it will be fixed soon.
